File: sdks/node/src/domain/resolver/service.ts
import type { ICyanResolver } from '../core/cyan_script.js';
import type { ResolverInput } from './input.js';
import type { ResolverOutput } from './output.js';

class ResolverService {
  private _resolver: ICyanResolver;

  constructor(resolver: ICyanResolver) {
    this._resolver = resolver;
  }

  async resolve(input: ResolverInput): Promise<ResolverOutput> {
    return await this._resolver.resolve(input);
  }
}

export { ResolverService };

File: sdks/python/cyanprintsdk/domain/resolver/service.py
from cyanprintsdk.domain.core.cyan_script import ICyanResolver
from cyanprintsdk.domain.resolver.input import ResolverInput
from cyanprintsdk.domain.resolver.output import ResolverOutput


class ResolverService:
    def __init__(self, resolver: ICyanResolver):
        self._resolver = resolver

    async def resolve(self, i: ResolverInput) -> ResolverOutput:
        return await self._resolver.resolve(i)
